### PR TITLE
chore: Upgrade React and Enzyme

### PR DESCRIPTION
**Pull request by lynnagara**

### CODE REVIEW

**Comment by billyvg** [comment]
Nice, tests are passing!

We should do a grep for `React.PropTypes` because I think I've seen a few places where its called. Should also check getsentry + make a tracking PR on getsentry to see if things break there.

File: tests/js/spec/views/__snapshots__/groupSimilarView.spec.jsx.snap [hunk 1]
```diff
@@ -38,1045 +38,18 @@ exports[`Issues Similar View renders with mocked data 1`] = `
...
-              ],
-            },
-          },
-        ]
-      }
-      onMerge={[Function]}
-      orgId="orgId"
-      projectId="projectId"
-    >
+    <LoadingIndicator>
```
Comment (MaxBittker) [inline tests/js/spec/views/__snapshots__/groupSimilarView.spec.jsx.snap]: what happened here?

File: tests/js/spec/views/__snapshots__/groupMergedView.spec.jsx.snap [hunk 2]
```diff
@@ -42,1297 +42,18 @@ exports[`Issues -> Merged View renders with mocked data 1`] = `
...
-            "state": "unlocked",
-          },
-        ]
-      }
-      onToggleCollapse={[Function]}
-      onUnmerge={[Function]}
-      orgId="orgId"
-      projectId="projectId"
-    >
+    <LoadingIndicator>
```
Comment (MaxBittker) [inline tests/js/spec/views/__snapshots__/groupMergedView.spec.jsx.snap]: was this intentional?

**Review by MaxBittker — CHANGES_REQUESTED** [review]
sweeeeeeet 💯 

+1 on a getsentry tracking PR! also, i noticed two snapshots that got a lot shorter, possibly due to some async render condition?

File: tests/js/spec/views/__snapshots__/groupSimilarView.spec.jsx.snap (hunk 1 above)
Comment (lynnagara) [reply tests/js/spec/views/__snapshots__/groupSimilarView.spec.jsx.snap]: Good pick up, thanks!

**Review by MaxBittker — APPROVED** [review]
💥 👍  love it

**Pull request merged**